backup.py

```diff
@@ -0,0 +1,142 @@
+    logging.info("Sending test email")
+    sendEmail("Test email",subject_tag="TEST")
+    exit(0)
+
+
+logging.debug("Starting Borg Backup")
+
+if not os.path.isdir(BORG_REPO):
+    logging.debug("Starting Repository Initialization")
+    os.mkdir(BORG_REPO)
+
+    command = ['borg', 'init', '--encryption=repokey']
+    logging.debug("Executing command: " + str(command))
+    try:
+        output = subprocess.run(command, shell=False,capture_output=True, check=True)
+    except subprocess.CalledProcessError as error:
+        logging.fatal("There was a problem initializing the repository")
+        sendEmail("Error during repository initialization, borg said: " + error.stderr, subject_tag="ERROR_BORG")
+        raise Exception("Error running command: " + error.stderr)
+    logging.debug("Ending Repository Initialization")
+
+
+borg_pid = subprocess.run(['pidof','borg'],shell=False,capture_output=True)
+if not borg_pid.stdout:
+    logging.debug("Clearing Locks; Borg Not Running")
+    subprocess.run(['borg','break-lock'],shell=False,check=True)
+
+
+logging.debug("Starting Daily Archive")
+timestamp = strftime("%Y-%m-%d-%s", localtime())
+command = ['borg','create','::'+timestamp,'/data']
+if BORG_CUSTOM_ARGS:
+    command[2:0] = BORG_CUSTOM_ARGS
+
+logging.debug("Executing command: " + str(command))
+try:
+    output = subprocess.run(command,shell=False,check=True,capture_output=True)
+except subprocess.CalledProcessError as error:
+    logging.fatal("There was a problem creating the daily archive")
+    sendEmail("Error during archive creation, borg said: " + error.stderr, subject_tag="ERROR_BORG")
+    raise Exception("Error running command: " + error.stderr)
+
+
+if BACKUP_PRUNE:
+    logging.debug("Starting Prune")
+    BACKUP_PRUNE=BACKUP_PRUNE.split()
+    command = ['borg','prune']
+    command+=BACKUP_PRUNE
+    try:
+        output = subprocess.run(command,shell=False,check=True,capture_output=True)
+    except subprocess.CalledProcessError as error:
+      logging.warning("There was a problem pruning the daily archive")
+      sendEmail("Error during archive prune: " + error.stderr, subject_tag="ERROR_PRUNE")
+      raise Exception("Error running command: " + error.stderr)
+    logging.debug("Ending Prune")
+
+
+logging.debug("Ending Daily Archive")
+logging.debug("Ending Borg Backup")
+logging.debug("Starting Rclone")
+
+command = ['rclone','sync','--transfers','16',BORG_REPO,BACKUP_LOCATION]
+logging.debug("Executing command: " + str(command))
+try:
+    output = subprocess.run(command,shell=False,capture_output=True,check=True)
+except subprocess.CalledProcessError as error:
+      logging.fatal("There was a problem syncing the backup")
+      sendEmail("Error during rclone sync, rclone said: " + error.stderr, subject_tag="ERROR_RCLONE")
+      raise Exception("Error running command: " + error.stderr)
+logging.debug("Ending Rclone")
+logging.debug("Ending Backup")
+if EMAIL_ENABLED:
+    sendEmail("Backup was successful")
```

Comment:
Can we move the `if EMAIL_ENABLED` check into the `sendMail` function and invert it?
e.g. `if not EMAIL_ENABLED: return`

Otherwise for any given error there will be the actual error and an error attempting to send the email as well.